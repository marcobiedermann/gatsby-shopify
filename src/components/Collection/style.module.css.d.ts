declare const styles: {
  readonly collection: string;
  readonly collection__title: string;
};
export = styles;
