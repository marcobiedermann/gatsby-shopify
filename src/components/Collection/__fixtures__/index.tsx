import { CollectionProps } from '..';
import productPreview from '../../ProductPreview/__fixtures__';

const collection: CollectionProps = {
  description: '',
  products: [productPreview],
  shopifyId: '12345678-9012-3456-7890-123456789012',
  title: 'Collection Title',
};

export default collection;
