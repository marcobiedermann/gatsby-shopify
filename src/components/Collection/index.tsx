import React, { FC } from 'react';
import ProductPreview, { ProductPreviewProps } from '../ProductPreview';
import * as styles from './style.module.css';

export interface CollectionProps {
  description: string;
  products: ProductPreviewProps[];
  shopifyId: string;
  title: string;
}

console.log({ styles });

const Collection: FC<CollectionProps> = (props) => {
  const { description, products, title } = props;

  return (
    <ul className={styles.collection}>
      <li>
        <div>
          <h1 className={styles.collection__title}>{title}</h1>
          <p>{description}</p>
        </div>
      </li>
      {products.map((product) => (
        <li key={product.shopifyId}>
          <ProductPreview {...product} />
        </li>
      ))}
    </ul>
  );
};

export default Collection;
