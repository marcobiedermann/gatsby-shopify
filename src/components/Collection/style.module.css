.collection {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  list-style: none;
  margin: 0;
  padding: 0;
}

.collection li {
  background-color: #fff;
  box-shadow: 0 0 0 1px #ededed;
  display: grid;
  place-items: center;
}

.collection li:first-child {
  grid-column-end: span 2;
  grid-row-end: span 2;
}

.collection__title {
  font-size: 6rem;
  margin-bottom: 0.125em;
}
