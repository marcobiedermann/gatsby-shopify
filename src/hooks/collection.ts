import { graphql, useStaticQuery } from 'gatsby';

interface Image {
  src: string;
}

interface ShopifyCollectionQuery {
  description: string;
  handle: string;
  image: Image;
  shopifyId: string;
  title: string;
}

interface AllShopifyCollectionQuery {
  allShopifyCollection: {
    edges: {
      node: ShopifyCollectionQuery;
    }[];
  };
}

export function useAllShopifyCollection() {
  return useStaticQuery<AllShopifyCollectionQuery>(graphql`
    {
      allShopifyCollection(sort: { fields: [title] }) {
        edges {
          node {
            description
            handle
            image {
              src
            }
            shopifyId
            title
          }
        }
      }
    }
  `);
}
