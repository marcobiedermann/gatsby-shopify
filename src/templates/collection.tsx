import React, { FC } from "react"
import Layout from "../components/Layout"

export interface CollectionTemplateProps {
  pageContext: {
    collection: {
      title: string
      description: string
    }
  }
}

const CollectionTemplate: FC<CollectionTemplateProps> = props => {
  const { pageContext } = props
  const { collection } = pageContext

  return (
    <Layout>
      <h1>{collection.title}</h1>
      <div>{collection.description}</div>
    </Layout>
  )
}

export default CollectionTemplate
