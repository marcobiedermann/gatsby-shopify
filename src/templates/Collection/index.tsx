import { graphql, Link, PageProps } from 'gatsby';
import React, { FC } from 'react';
import Collection from '../../components/Collection';
import Grid from '../../components/Grid';
import Layout from '../../components/Layout';

interface Product {
  handle: string;
  shopifyId: string;
  title: string;
}

interface DataProps {
  shopifyCollection: {
    description: string;
    products: Product[];
    shopifyId: string;
    title: string;
  };
}

const CollectionTemplate: FC<PageProps<DataProps>> = (props) => {
  const {
    data: { shopifyCollection },
    location,
  } = props;

  return (
    <Layout location={location}>
      <Grid>
        <Collection {...shopifyCollection} />
        <ul>
          {shopifyCollection.products.map((product) => (
            <li key={product.shopifyId}>
              <h2>
                <Link to={`/products/${product.handle}`}>{product.title}</Link>
              </h2>
            </li>
          ))}
        </ul>
      </Grid>
    </Layout>
  );
};

export const pageQuery = graphql`
  query CollectionByShopifyId($shopifyId: String!) {
    shopifyCollection(shopifyId: { eq: $shopifyId }) {
      description
      products {
        handle
        shopifyId
        title
      }
      shopifyId
      title
    }
  }
`;

export default CollectionTemplate;
