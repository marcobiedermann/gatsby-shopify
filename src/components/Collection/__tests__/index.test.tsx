import { render } from '@testing-library/react';
import React from 'react';
import Collection from '..';
import collection from '../__fixtures__';

describe('collection component', () => {
  it('should render correctly', () => {
    const { container } = render(<Collection {...collection} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
