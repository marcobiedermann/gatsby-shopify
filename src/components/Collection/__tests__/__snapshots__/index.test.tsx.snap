// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collection component should render correctly 1`] = `
<ul>
  <li>
    <div>
      <h1>
        Collection Title
      </h1>
      <p />
    </div>
  </li>
  <li>
    <div>
      <figure>
        <a
          href="/products/product"
        >
          <img
            alt="Product Title"
            src="/path/to/image.jpg"
          />
        </a>
      </figure>
      <h3>
        <a
          href="/products/product"
        >
          Product Title
        </a>
      </h3>
      <p>
        100
         
        EUR
      </p>
    </div>
  </li>
</ul>
`;
